extensions/amp-story/1.0/media-tasks.js

```diff
@@ -109,200 +108,132 @@ function copyAttributes(fromEl, toEl) {
 }
 
 /**
- * Base class for tasks executed in order on HTMLMediaElements.
+ * A task executed in order on HTMLMediaElements.
+ * @interface
  */
 export class MediaTask {
-  /**
-   * @param {string} name
-   * @param {!Object=} options
-   */
-  constructor(name, options = {}) {
-    /** @private @const {string} */
-    this.name_ = name;
-
-    const deferred = new Deferred();
```

Comment:
Delayed promise resolution is now handled entirely in `enqueueMediaElementTask_`.